source/retryable-writes/retryable-writes.rst

```diff
@@ -229,6 +228,8 @@ user.
 Supported Write Operations
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+MongoDB 3.6 will support retryability for some, but not all, write operations.
```

Comment:
> When retrying a write command, drivers MUST resend the command with the same transaction ID. Drivers MAY resend the original wire protocol message (see: Can drivers resend the same wire protocol message on retry attempts?). If the second attempt also fails, drivers MUST raise its corresponding error to the user.

In practice, the original error could only be a NetworkException. Since server selection occurs between the first and second attempt and we may have selected a new primary server, what value is there in throwing the original NetworkException? It is likely to reflect the current state of the system. Consider:

 * Node A went offline and the replica set fails over to node B
 * Driver, using outdated topology information, sends a write to A and receives a NetworkException
 * Driver performs server selection and realizes that B is the new primary
 * Node A recovers and is reelected and B goes offline
 * Driver retries the write with B and receives a NetworkException

If B never went offline, I suppose we'd get a server response that it's not a master and simply throw that to the user; however, if it is offline and we get a NetworkException, I think that's more relevant to the user than the previous NetworkException for A. Perhaps this is all a moot point because an election would never complete in the time between server selection and the retry attempt.

Either way, I should clarify that "fails" in "If the second attempt also fails" is only referring to another NetworkException. If the retry attempt receives an `{ ok: 0 }` response from the server indicating that the command failed (e.g. duplicate key exception), we obviously want to throw _that_ to the user instead of the original NetworkException.